### ISSUE TITLE
reduce function not recognized

### ISSUE THREAD

**Issue by Salil999** [PyCQA/pylint#1277]
### Steps to reproduce
1. Create blank file with following code:
```python
x = [x for x in range(10)]
print(reduce(lambda a, b: a + b, x))
```

### Current behavior
I have attached a screenshot for clarity. Basically, the "reduce" function is not recognized.
![image](https://cloud.githubusercontent.com/assets/8891729/21875326/cc0be9ce-d840-11e6-9754-2c6d0d0ba3a6.png)

### Expected behavior
I'm hoping that it will not give me a warning for that. However, I am using [this extension](https://github.com/DonJayamanne/pythonVSCode) that is supposed to include pylint, so I'm not sure if it is a problem on their end or on this end. But it seemed more related to pylint, so I made an issue here.

### pylint --version output
```text
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  5 2015, 21:12:44) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

### pylint script.py output
```text
************* Module temp
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Invalid constant name "x" (invalid-name)
E:  2, 6: Undefined variable 'reduce' (undefined-variable)


Report
======
2 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |1      |1          |=          |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |0      |0          |=          |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|method   |0      |0          |=          |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|function |0      |0          |=          |0           |0        |
+---------+-------+-----------+-----------+------------+---------+



Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |3      |75.00 |3        |=          |
+----------+-------+------+---------+-----------+
|docstring |0      |0.00  |0        |=          |
+----------+-------+------+---------+-----------+
|comment   |0      |0.00  |0        |=          |
+----------+-------+------+---------+-----------+
|empty     |1      |25.00 |1        |=          |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |=          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |2      |2        |=          |
+-----------+-------+---------+-----------+
|refactor   |0      |0        |=          |
+-----------+-------+---------+-----------+
|warning    |0      |0        |=          |
+-----------+-------+---------+-----------+
|error      |1      |1        |=          |
+-----------+-------+---------+-----------+



Messages
--------

+-------------------+------------+
|message id         |occurrences |
+===================+============+
|undefined-variable |1           |
+-------------------+------------+
|missing-docstring  |1           |
+-------------------+------------+
|invalid-name       |1           |
+-------------------+------------+



Global evaluation
-----------------
Your code has been rated at -25.00/10 (previous run: -25.00/10, +0.00)
```

**Comment by thisch** [comment]
On python3, ``reduce`` is no longer a builtin command. Therefore, you get the error.

http://stackoverflow.com/questions/10226381/nameerror-global-name-reduce-is-not-defined

**Comment by PCManticore** [comment]
@thisch is right, ``reduce`` lives in ``functools`` in Python 3.
